src/components/controls/Button.tsx

```diff
@@ -35,16 +36,28 @@ class Button extends React.Component<IButtonProps, {}> {
         }
     }
 
-    private getClassname = (type: ButtonType, className?: string) => {
+    private getClassname = (type: ButtonType, disabled: boolean, className?: string) => {
         const typeClass = this.getTypeClass(type);
-        return classNames(s.button, typeClass, className);
+        return classNames(s.button, typeClass, className, disabled ? s.disabled : null);
+    }
+
+    private onClick = (e: any) => {
+        if (!this.props.disabled) {
+            this.props.onClick(e);
+        }
     }
 
     public render(): any {
         return (
             <div
-                className={this.getClassname(this.props.type, this.props.className)}
-                onClick={this.props.onClick}
+                className={
+                    this.getClassname(
+                        this.props.type,
+                        this.props.disabled!,
```

Comment:
Use this.props.disabled at getClassName(), no need to pass here as param

Reply:
Personally, I think the scope of functions is better visualized if we try to pass used as variables as parameters, instead of use them in the function directly from props.

from the 'this.getClassname(a,b,c)' I directly see on what basis the function makes its decision. 
If we would have 'this.getClassname()', we know it return something, but we don't know on what basis.

Note, I don't think any answer to this is wrong or correct, this is just my personal preference.

Reply:
How about using this code:
```
                className={
                    classnames(
                        s.button,
                        this.props.className,
                        this.getTypeClass(this.props.type),
                        this.props.disabled ? s.disabled : null,
                    )
                }
```

and remove getClassname method.

Reply:
Fixed already.